osf/models/files.py

```diff
@@ -445,7 +445,7 @@ def update(self, revision, data, user=None):
                 break
         else:
             # Insert into history if there is no matching etag
-            if data.get('modified'):
+            if data.get('modified') and None not in [item['modified'] for item in self.history]:
```

Comment:
So... just to see if I understand this correctly: I should do something along the lines of
```
If no matching etags;
    append revision to history

if in dev_mode:
  if update() is called by a non GithubFile instance:
    log to sentry with distinctive title
```